presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcSelectivePageSource.java

```diff
@@ -114,7 +114,9 @@ public Page getNextPage()
             Page page;
             if (supplyRowIDs) {
                 page = recordReader.getNextPage(true);
-                page = fillInRowIDs(page);
+                if (page != null) {
```

Comment:
let's remove this check because `fillInRowIDs` does the same

Reply:
done